frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/services/ReverificationResultService.java

```diff
@@ -0,0 +1,183 @@
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.impl.ECDSA;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.oauth2.sdk.AuthorizationCode;
+import com.nimbusds.oauth2.sdk.AuthorizationCodeGrant;
+import com.nimbusds.oauth2.sdk.TokenRequest;
+import com.nimbusds.oauth2.sdk.TokenResponse;
+import com.nimbusds.oauth2.sdk.auth.JWTAuthenticationClaimsSet;
+import com.nimbusds.oauth2.sdk.auth.PrivateKeyJWT;
+import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import com.nimbusds.oauth2.sdk.id.Audience;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.JWTID;
+import com.nimbusds.openid.connect.sdk.UserInfoRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.kms.model.SignRequest;
+import software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec;
+import uk.gov.di.authentication.shared.exceptions.JwtParseException;
+import uk.gov.di.authentication.shared.exceptions.TokenRequestException;
+import uk.gov.di.authentication.shared.exceptions.TokenResponseException;
+import uk.gov.di.authentication.shared.exceptions.UnsuccessfulReverificationResponseException;
+import uk.gov.di.authentication.shared.helpers.ConstructUriHelper;
+import uk.gov.di.authentication.shared.helpers.NowHelper;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.KmsConnectionService;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+public class ReverificationResultService {
+    private static final Logger LOG = LogManager.getLogger(ReverificationResultService.class);
+    private final ConfigurationService configurationService;
+    private final KmsConnectionService kmsConnectionService;
+    private static final Long CLIENT_ASSERTION_LIFETIME = 5L;
+    private static final JWSAlgorithm TOKEN_ALGORITHM = JWSAlgorithm.ES256;
+
+    public ReverificationResultService(
+            ConfigurationService configurationService, KmsConnectionService kmsConnectionService) {
+        this.configurationService = configurationService;
+        this.kmsConnectionService = kmsConnectionService;
+    }
+
+    public ReverificationResultService(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+        this.kmsConnectionService = new KmsConnectionService(configurationService);
+    }
+
+    public TokenResponse getToken(String authCode) {
+        var tokenRequest = constructTokenRequest(authCode);
+        return sendTokenRequest(tokenRequest);
+    }
+
+    public TokenRequest constructTokenRequest(String authCode) {
```

Comment:
Are these methods public just for testing? I'm not going to be militant about it, but that feel like smell to me